Index: src/metapensiero/pj/transformations/obvious.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/metapensiero/pj/transformations/obvious.py	(revision e8a9727d87f0b83af669481c51a773fe98d73efc)
+++ src/metapensiero/pj/transformations/obvious.py	(revision 3ad8eec51d69ce5a69d76e229d179994dcdf8897)
@@ -278,6 +278,18 @@
     return cls()
 
 
+# Take care of Python 3.8's deprecations:
+# https://docs.python.org/3/library/ast.html#node-classes
+def Constant(t, x):
+    if isinstance(x.value, bool) or x.value is None:
+        return NameConstant(t, x)
+    elif isinstance(x.value, (int, float, complex)):
+        return Num(t, x)
+    elif isinstance(x.value, str):
+        return Str(t, x)
+    else:
+        raise ValueError('Unknown data type received.')
+
 def Yield(t, x):
     return JSYield(x.value)
 
